src/GafferVDB/PointsGridToPoints.cpp

```diff
@@ -0,0 +1,557 @@
+	if ( itConverter == converters.end() )
+	{
+		return;
+	}
+
+	IECoreScene::PrimitiveVariable primVar;
+
+	auto primVarIt = variableMap.find( name );
+	if( primVarIt == variableMap.end() )
+	{
+		primVar = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, itConverter->second.m_create( count ) );
+		variableMap[name] = primVar;
+	}
+	else
+	{
+		primVar = primVarIt->second;
+	}
+
+	itConverter->second.m_append( primVar.data.get(), arrayData, leafIter);
+}
+
+IECoreScene::PointsPrimitivePtr createPointsPrimitive( openvdb::GridBase::ConstPtr baseGrid, std::function<bool( const std::string & )> primitiveVariableFilter )
+{
+	openvdb::points::PointDataGrid::ConstPtr pointsGrid = openvdb::GridBase::constGrid<openvdb::points::PointDataGrid>( baseGrid );
+	if( !pointsGrid )
+	{
+		return nullptr;
+	}
+
+	openvdb::Index64 count = openvdb::points::pointCount( pointsGrid->tree() );
+
+	IECore::V3fVectorDataPtr pointData = new IECore::V3fVectorData();
+	auto &points = pointData->writable();
+	points.reserve( count );
+
+	IECoreScene::PrimitiveVariableMap primVars;
+
+	for( auto leafIter = pointsGrid->tree().cbeginLeaf(); leafIter; ++leafIter )
+	{
+		const openvdb::points::AttributeArray &array = leafIter->constAttributeArray( "P" );
+		openvdb::points::AttributeHandle<openvdb::Vec3f> positionHandle( array );
+
+		for( auto indexIter = leafIter->beginIndexOn(); indexIter; ++indexIter )
+		{
+			const openvdb::points::AttributeSet &attributeSet = leafIter->attributeSet();
+			const openvdb::points::AttributeSet::Descriptor &descriptor = attributeSet.descriptor();
+
+			for (const auto &it : descriptor.map() )
+			{
+				const std::string &attributeName = it.first;
+
+				if ( !primitiveVariableFilter( attributeName ) )
+				{
+					continue;
+				}
+
+				size_t index = it.second;
+
+				const openvdb::points::AttributeArray *attributeArray = attributeSet.get( index );
+				appendPrimitiveVariableData( primVars, attributeName, descriptor.type ( index ).first, leafIter, *attributeArray, count);
+			}
+
+			openvdb::Vec3f voxelPosition = positionHandle.get( *indexIter );
+			const openvdb::Vec3d xyz = indexIter.getCoord().asVec3d();
+			openvdb::Vec3f worldPosition = pointsGrid->transform().indexToWorld( voxelPosition + xyz );
+			points.push_back( Imath::Vec3<float>( worldPosition[0], worldPosition[1], worldPosition[2] ) );
+		}
+	}
+
+	IECoreScene::PointsPrimitivePtr newPoints = new IECoreScene::PointsPrimitive( pointData );
+
+	for ( auto it : primVars )
+	{
+		newPoints->variables[it.first] = it.second;
+	}
+
+	return newPoints;
+}
+
+} //namespace
+
+IE_CORE_DEFINERUNTIMETYPED( PointsGridToPoints );
+
+size_t PointsGridToPoints::g_firstPlugIndex = 0;
+
+PointsGridToPoints::PointsGridToPoints( const std::string &name ) : SceneElementProcessor( name )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+
+	addChild( new StringPlug( "grid", Plug::In, "points" ) );
+
+	// 'names' & 'invertNames' match PrimitiveVariableProcessor
```

Comment:
Do you think these are specific enough? In the PrimitiveVariableProcessor nodes the name of the node provides enough context to assume that the names are of primitive variables, but I'm less sure about that here. Still, it's easy enough to change a plug name later I guess...

Reply:
I think they could be a littel more specific and descriptive but I got stuck on what to call them. I did consider *attributes* & *invertAttributes* but then there's overlap with Primitive attributes though, so I just left it matching the PrimitiveVariable processor node.